src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -525,29 +525,23 @@ private void ReleaseUnmanagedResources()
         {
             batch.Dispose();
         }
-        
+
         _db?.Dispose();
     }
 
     private void Dispose(bool disposing)
     {
         if (!_isDisposed)
         {
-            // ReSharper disable once ConstantConditionalAccessQualifier
-            _logger?.Info($"Disposing DB {Name}");
-
             if (disposing)
             {
+                if (_logger.IsInfo) _logger.Info($"Disposing DB {Name}");
```

Comment:
Dispose(false) does nothing, right? Could be simplified if so

Reply:
Hm... I wonder because the `_rocksBatch.Dispose();` might do useful thing, WDYT?

Reply:
Yep, let batches be disposed, I meant we have Dispose(bool disposing), if `false` we just reset disposing flag, looks useless a bit

Reply:
On other side finalizer may help to raise some useful exceptions if we forget to dispose

Reply:
So maybe either on !disposing still dispose inner batches? Or add finalizers to batches? Which is better?

Reply:
`Dispose` is cool imho

Reply:
`~DbOnTheRocks` should not even ever be called if we control the lifetime properly. If `!disposing && !_isDisposed` we can notify of a problem and still try to clean up, yes.

Reply:
`~DbOnTheRocks` doesn't make sense, the wrapper underneath should have finalizer. The one with native IntPtr, not this one.

Reply:
`Dispose(false)` is never used now